modules/http/src/test/java/org/glassfish/grizzly/http/HttpResponseParseTest.java

```diff
@@ -114,17 +115,59 @@ public void testHeadersN() throws Exception {
         headers.put("Content-length", new Pair<String, String>("2345", "2345"));
         doHttpResponseTest("HTTP/1.0", 200, "DONE", headers, "\n");
     }
-    
+
     public void testDecoder100continueThen200() {
         try {
-            doTestDecoder("HTTP/1.1 100 Continue\n\nHTTP/1.1 200 OK\n\n", 4096);
-            assertTrue(true);
+            HttpPacket packet = doTestDecoder("HTTP/1.1 100 Continue\n\nHTTP/1.1 200 OK\n\n", 4096);
+            assertTrue(packet.getHttpHeader() instanceof HttpResponsePacket);
+            HttpResponsePacket response = (HttpResponsePacket) packet.getHttpHeader();
+            assertTrue(response.getStatus() == 200);
         } catch (IllegalStateException e) {
             logger.log(Level.SEVERE, "exception", e);
             assertTrue("Unexpected exception", false);
         }
     }
 
+    public void testDecoder100continueWithAHeaderThen200() {
+        try {
+            HttpPacket packet = doTestDecoder("HTTP/1.1 100 Continue\r\nConnection: keep-alive\r\n\r\nHTTP/1.1 200 OK\r\n\r\n", 4096);
+            assertTrue(packet.getHttpHeader() instanceof HttpResponsePacket);
+            HttpResponsePacket response = (HttpResponsePacket) packet.getHttpHeader();
+            assertTrue(response.getStatus() == 200);
+        } catch (IllegalStateException e) {
```

Comment:
I would remove this catch-block, add the throws for the method and let the test fail with the unexpected exception. I would change that in the previous test that uses this too.